jdaviz/configs/default/plugins/viewers.py

```diff
@@ -362,3 +392,413 @@ def _ref_or_id(self):
     def set_plot_axes(self):
         # individual viewers can override to set custom axes labels/ticks/styling
         return
+
+
+@viewer_registry("jdaviz-profile-viewer", label="Profile 1D")
+class JdavizProfileView(JdavizViewerMixin, BqplotProfileView):
+    # categories: zoom resets, zoom, pan, subset, select tools, shortcuts
+    tools_nested = [
+                    ['jdaviz:homezoom', 'jdaviz:prevzoom'],
+                    ['jdaviz:boxzoom', 'jdaviz:xrangezoom', 'jdaviz:yrangezoom'],
+                    ['jdaviz:panzoom', 'jdaviz:panzoom_x', 'jdaviz:panzoom_y'],
+                    ['bqplot:xrange'],
+                    ['jdaviz:sidebar_plot', 'jdaviz:sidebar_export']
+                ]
+
+    default_class = NDDataArray
+    _state_cls = FreezableProfileViewerState
+    _default_profile_subset_type = None
+
+    def __init__(self, *args, **kwargs):
+        default_tool_priority = kwargs.pop('default_tool_priority', [])
+        super().__init__(*args, **kwargs)
+
+        self._subscribe_to_layers_update()
+        self.initialize_toolbar(default_tool_priority=default_tool_priority)
+        self._offscreen_lines_marks = OffscreenLinesMarks(self)
+        self.figure.marks = self.figure.marks + self._offscreen_lines_marks.marks
+
+        self.state.add_callback('show_uncertainty', self._show_uncertainty_changed)
+
+        self.display_mask = False
+
+        # Change collapse function to sum
+        default_collapse_function = kwargs.pop('default_collapse_function', 'sum')
+
+        self.state.function = default_collapse_function
+
+    def _expected_subset_layer_default(self, layer_state):
+        super()._expected_subset_layer_default(layer_state)
+
+        layer_state.linewidth = 3
+
+    def data(self, cls=None):
+        # Grab the user's chosen statistic for collapsing data
+        statistic = getattr(self.state, 'function', None)
```

Comment:
I don't think this auto-collapsing logic is used in cubeviz anymore... is it needed for Rampviz?